core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -330,7 +330,7 @@ private[spark] object JettyUtils extends Logging {
 
           // redirect the HTTP requests to HTTPS port
           httpConnector.setName(REDIRECT_CONNECTOR_NAME)
-          collection.addHandler(createRedirectHttpsHandler(securePort, scheme))
+          collection.addHandler(createRedirectHttpsHandler(connector, scheme))
```

Comment:
Ah, good point. I guess I was thinking about the code in master which starts the connector right away.